blueprint/cli/src/bin/db.rs

```diff
@@ -242,3 +291,117 @@ async fn get_root_db_client(config: &DatabaseConfig) -> PgConnection {
 
     connection
 }
+
+fn get_cargo_path() -> Result<String, anyhow::Error> {
+    std::env::var("CARGO")
+        .map_err(|_| anyhow!("Please invoke me using Cargo, e.g.: `cargo db <ARGS>`"))
+}
+
+/// Ensure that the correct version of sqlx-cli is installed,
+/// and install it if it isn't.
+async fn ensure_sqlx_cli_installed(ui: &mut UI<'_>) -> Result<(), anyhow::Error> {
+    macro_rules! sqlx_cli_version {
+        ($vers:literal) => {
+            /// The version of sqlx-cli required
+            const SQLX_CLI_VERSION: &str = $vers;
+            /// The expected version output of sqlx-cli
+            const SQLX_CLI_VERSION_STRING: &[u8] = concat!("sqlx-cli-sqlx ", $vers).as_bytes();
+        };
+    }
+    sqlx_cli_version!("0.8.2");
```

Comment:
I think it should check just for the presence of sqlx-cli, there might be no good reason to force a specific version, because as I understand, somebody would have to specifically change the version here in the code directly.

Since we're already setting an SQLX_OFFLINE variable, maybe we should add SQLX_VERSION variable to the `.env` file and install a version specified there? This way it's at least easily configurable

Reply:
Yeah this is a bit of a design decision, whether or not we want the sqlx-cli version to be configurable. I do think it's going to be annoying if the user has a newer version installed. Maybe we should make it semver-aware

Reply:
Allowing the user to fully specify the version means that they would be allowed to install an incompatible version. Gerust cli is however only going to work with senver-compatible versions of sqlx-cli. Therefore I updated the code such that:
- it attempts to install the latest compatible version of sqlx-cli if none is installed or if an incompatible version is installed
- use currently installed version sqlx-cli in case it's compatible with what gerust requires (`^0.8` in this case)

I thing this is still quite foolproof, and does not get too annoying if you're using a specific version of sqlx-cli for some other project.

Reply:
I think that works :+1:  although I'm not sure why it's important to force a 0.8 version.
Following semver rules should allow us to use any version that's >0.8 because 0.9, 0.10 and so on should support the same features and only 1.0 could actually be breaking.

Either way a worst case scenario somebody will just edit the version requirement to whatever version they want.

Reply:
@BobrImperator 
> Following semver rules should allow us to use any version that's >0.8 because 0.9, 0.10 and so on should support the same features and only 1.0 could actually be breaking.

It's a bit more complicated than that: 

> This guide uses the terms “major” and “minor” assuming this relates to a “1.0.0” release or later. Initial development releases starting with “0.y.z” can treat changes in “y” as a major release, and “z” as a minor release. “0.0.z” releases are always major changes. **This is because Cargo uses the convention that only changes in the left-most non-zero component are considered incompatible.** ([source](https://doc.rust-lang.org/cargo/reference/semver.html))

Emphasis mine. Meaning that a 0.9.x release might involve breaking changes. So it's best to pin the to major version at the very least.

It may make sense to also pin the minor version number so as to keep in sync with the `sqlx` library crate used in the project, but that would require users to sacrifice flexibility, so let's keep it at pinning the major version.